SEP-38: fixes and improvements (#987)
stellar/stellar-protocol: ecosystem/sep-0038.md

Track: Standard
Status: Draft
Created: 2021-04-09
Updated: 2021-06-28
Discussion: https://github.com/stellar/stellar-protocol/issues/901

> Version 1.1.3

Reviewer: It looks like this change adds fields to the responses as new functionality, so this version should be a minor version. i.e. `1.2.0`

Author: https://github.com/stellar/stellar-protocol/pull/987/commits/a727baf7a92e2e581a3fbd9c7295759b3a741605

      "delivery_methods": [
        {
          "name": "cash",

>           "description": "Deposit cash BRL to one of our payout locations."

Reviewer: Payout = withdraw, so is this actually a deposit or is this a withdrawal? Payout might be the wrong term.

Author: How does this sound? https://github.com/stellar/stellar-protocol/pull/987/commits/023316ccd06f76af962074c941e0c7db148c1d17

This endpoint can be used to fetch the [indicative](https://www.investopedia.com/terms/i/indicativequote.asp) prices of available off-chain assets in exchange for a Stellar asset and vice versa.

These prices are just indicatives provided at a given time, the actual price will be calculated according with the market price at conversion time, after the Anchor receives the funds from a User.

> It is expected that the prices returned **include a margin for the provider as a service fee**, and this margin will vary depending on the directional flow of funds.

Reviewer: Can you remove the emphasis (bold)? SEPs contain a lot of important information and to bold one piece of information over the other implies there's areas of text that can be ignore.

Author: https://github.com/stellar/stellar-protocol/pull/987/commits/a727baf7a92e2e581a3fbd9c7295759b3a741605

The client must provide either `sell_amount` or `buy_amount`, but not both.

> These prices are just indicatives provided at a given time, the actual price will be calculated according with the market price at conversion time, after the Anchor receives the funds from a User.

Reviewer: ```suggestion
These prices are indicative. The actual price will be calculated at conversion time, once the Anchor receives the funds from a User.
```

Author: https://github.com/stellar/stellar-protocol/pull/987/commits/a727baf7a92e2e581a3fbd9c7295759b3a741605

> It is expected that the price returned **includes a margin for the provider as a service fee**, and this margin will vary depending on the directional flow of funds.

Reviewer: Same comment regarding emphasis.

Author: https://github.com/stellar/stellar-protocol/pull/987/commits/a727baf7a92e2e581a3fbd9c7295759b3a741605

#### Transaction Fees

> It is expected for the quote returned to **include a margin for the provider as a service fee**, and this margin will vary depending on the directional flow of funds and the delivery method.

Reviewer: ```suggestion
The quote returned includes any margin that the provider may keep as a service fee, and this margin may vary depending on the directional flow of funds, the delivery method, amount, or other factors.
```

Author: https://github.com/stellar/stellar-protocol/pull/987/commits/972c678112de0a7ac05ce7d43ba9a9b220c55511

`id` | string | The `id` specified in the request.
`expires_at` | [UTC ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) string | The date and time by which the anchor must receive funds from the client.
`price` | string | The price offered by the anchor for one unit of `buy_asset` in terms of `sell_asset`. In traditional finance, `buy_asset` would be referred to as the base asset and `sell_asset` as the counter asset.
`buy_amount` | string | The amount of `buy_asset` to be exchanged for `sell_asset`. `price * buy_amount = sell_amount` must be true up to the number of decimals required for `buy_asset`.
`sell_asset` | string | The asset the client would like to sell. Ex. `USDC:G...`, `iso4217:ARS`
`sell_amount` | string | The amount of `sell_asset` to be exchanged for `buy_asset`.
`buy_asset` | string | The asset the client would like to exchange for `sell_asset`.
`buy_amount` | string | The amount of `buy_asset` to be exchanged for `sell_asset`. `price * buy_amount = sell_amount` must be true up to the number of decimals required for `buy_asset`.

> ##### Examples

Reviewer 2: Can we keep the examples deleted examples but just move it to the `GET /price` endpoint?

Author: There are already plenty of examples for the `GET /price` endpoint [here](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0038.md#examples-1).

The examples I removed were using `GET /price` requests but showing responses with fields from `GET /quote`, so they wouldn't fit any of those endpoints.
